src/display/draw_layer.js

```diff
@@ -0,0 +1,203 @@
+/* Copyright 2023 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DOMSVGFactory } from "./display_utils.js";
+import { shadow } from "../shared/util.js";
+
+/**
+ * Manage the SVGs drawn on top of the page canvas.
+ * It's important to have them directly on top of the canvas because we want to
+ * be able to use mix-blend-mode for some of them.
+ */
+class DrawLayer {
+  #parent = null;
+
+  #id = 0;
+
+  #mapping = new Map();
+
+  constructor({ pageIndex }) {
+    this.pageIndex = pageIndex;
+  }
+
+  setParent(parent) {
```

Comment:
The `DrawLayer` is a bit mis-named since it isn't really a layer but it's an helper to add some SVGs under its parent element.
I firstly thought having a real layer with different objects to draw inside, but because of the mix-blend-mode which is different for the highlight itself (e.g. `multiply`) and for the outline (e.g. `normal`), I had to move the different objects into different SVG nodes.
That said, we want to apply the mix-blend-mode when composing a SVG with the canvas, hence the SVG has to be a child of the canvas wrapper which is changed when zooming or rotating:
https://github.com/mozilla/pdf.js/blob/d8424a43baf4c18dd19c77f466bbfff2b15cd8f4/web/pdf_page_view.js#L851